hermes/bin/src/runtime_extensions/hermes/cron/mod.rs

```diff
@@ -1,15 +1,582 @@
+    let (hour, minute, _secs) = delayed.to_hms();
+    let when = mkcron_impl(
+        &vec![],
+        &vec![CronComponent::At(month)],
+        &vec![CronComponent::At(day)],
+        &vec![CronComponent::At(hour)],
+        &vec![CronComponent::At(minute)],
+    );
+    Ok(CronTagged { when, tag })
+}
+
+/// Convert `CronTime` arguments to a `CronSched`.
+pub(crate) fn mkcron_impl(
+    dow: &CronTime, month: &CronTime, day: &CronTime, hour: &CronTime, minute: &CronTime,
+) -> CronSched {
+    let dow_schedule: CronSched = cron_time_to_cron_sched_dow(dow);
+    let month_schedule: CronSched = cron_time_to_cron_sched_month(month);
+    let day_schedule: CronSched = cron_time_to_cron_sched_day(day);
+    let hour_schedule: CronSched = cron_time_to_cron_sched_hour(hour);
+    let minute_schedule: CronSched = cron_time_to_cron_sched_minute(minute);
+    // Return the merged schedule.
+    format!("{minute_schedule} {hour_schedule} {day_schedule} {month_schedule} {dow_schedule}",)
+}
+
+/// Convert a `CronTime` to a `CronSched`.
+///
+/// Silently clamps values within the specified `min_val..=max_val` range, removes
+/// duplicates, merges overlaps, and ensures that range values are in the right order:
+/// `first <= last`.
+///
+/// If the `CronTime` contains no components, returns `*`.
+/// If the `CronTime` contains `CronComponent::All`, returns `*`.
+/// If the `CronTime` contains `CronComponent::Range(first, last)`, returns `*`.
+/// If the `CronTime` contains overlapping components, it merges them.
+fn cron_time_to_cron_sched(cron_time: &CronTime, min_val: u8, max_val: u8) -> CronSched {
+    // If vec has no components or if it includes `CronComponent::All`, skip processing and
+    // return "*"
+    let cron_sched: CronSched = if cron_time.is_empty() {
+        format!("{}", CronComponent::All)
+    } else {
+        // Silently clamp values, and ensure that range values are in the right order: `first <=
+        // last`. For the case of finding `CronComponent::Range((final, last))`, it is
+        // replaced with `CronComponent::All`.
+        let clamped: Vec<CronComponent> = clamp_cron_time_values(cron_time, min_val, max_val);
+        // Merge overlapping components
+        let merged: Vec<CronComponent> = merge_cron_time_overlaps(clamped);
+        // Return the merged cron schedule
+        merged
+            .into_iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<CronSched>>()
+            .join(",")
+    };
+    cron_sched
+}
+
+/// Clamp values within the specified `min_val..=max_val` range
+fn clamp_cron_time_values(
+    cron_time: &[CronComponent], min_val: u8, max_val: u8,
+) -> Vec<CronComponent> {
+    cron_time
+        .iter()
+        .map(|d| d.clamp_inner(min_val, max_val))
+        .collect()
+}
+
+/// Scan over the remaining components and merge them if they overlap
+fn merge_cron_time_overlaps(cron_time: Vec<CronComponent>) -> Vec<CronComponent> {
```

Comment:
Just as a suggestion, does it make sense to replace `Vec<CronComponent>` with the `BTreeSet` so it will no need to `sort` everytime.

Reply:
Yeah, it's cleaner, thanks.